var/spack/repos/builtin/packages/roctracer-dev/package.py

```diff
@@ -17,12 +17,17 @@ class RoctracerDev(CMakePackage):
 
     maintainers = ['srekolam', 'arjun-raj-kuppala']
 
+    version('3.8.0', sha256='5154a84ce7568cd5dba756e9508c34ae9fc62f4b0b5731f93c2ad68b21537ed1')
     version('3.7.0', sha256='6fa5b771e990f09c242237ab334b9f01039ec7d54ccde993e719c5d6577d1518')
     version('3.5.0', sha256='7af5326c9ca695642b4265232ec12864a61fd6b6056aa7c4ecd9e19c817f209e')
 
     depends_on('cmake@3:', type='build')
+    depends_on('py-setuptools@44.1.0', type='build')
+    depends_on('python@2.7.18', type='build')
```

Comment:
It doesn't work with any other version of Python or setuptools??

Reply:
Not really, i mean i need python2 version for sure but i ran into below issue if i dont select above specified version for setuptools
-----------------------------------------------------------
_```
===> Error: An unsatisfiable version constraint has been detected for spec:

python@2.7.18%gcc@7.3.1+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib arch=linux-centos7-zen
while trying to concretize the partial spec:
    py-setuptools@50.1.0%gcc@7.3.1 arch=linux-centos7-zen
py-setuptools requires python version 3.5:, but spec asked for 2.7.18
```_

Reply:
That's a bug in the concretizer that should be fixed in the next release. Until then, it would be better to accept a range of possible versions, not just the latest version that supports Python 2.